src/main/java/app/linguistai/bmvp/service/gamification/UserStreakService.java

```diff
@@ -76,16 +76,15 @@ private RUserStreak checkUserStreakForUpdate(RUserStreak rUserStreak) throws Exc
     protected RUserStreak incrementUserStreak(UserStreak userStreak) throws Exception {
         try {
             // Assume user streak increment operation is correct
-            UserStreak newUserStreak = new UserStreak();
-            newUserStreak.setCurrentStreak(userStreak.getCurrentStreak() + 1);
-            newUserStreak.setHighestStreak(
+            userStreak.setCurrentStreak(userStreak.getCurrentStreak() + 1);
+            userStreak.setHighestStreak(
                 userStreak.getHighestStreak() <= userStreak.getCurrentStreak()
                 ? userStreak.getHighestStreak() + 1
```

Comment:
This part is kind of hard to read, I would probably prefer one of the following:

- Extracting this logic as a utility function
- Assigning the result of the ternary operation to a nicely named variable, then setting the highest streak